### PR TITLE
[NUI] Add TabItem to support TabView in xaml

### PR DESCRIPTION
**Pull request by jaehyun0cho**
To support TabView in xaml, TabItem has been added.

TabView provides Add(TabItem) to add a tab from the given TabItem.
TabItem contains Title, IconURL of a new TabButton in TabView's TabBar
and Content of a new View in TabView's TabContent.

### API Changes ###

 - ACR:

### CODE REVIEW

**Comment by jaehyun0cho** [comment]
This patch has been updated to use Add(TabItem).
By doing the above, TabItem does not need to inherit from View.

File: src/Tizen.NUI.Components/Controls/TabView.cs [hunk 1]
```diff
@@ -252,6 +252,38 @@ public void RemoveTab(int index)
...
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual void Add(TabItem tabItem)
+        {
+            var hasTitle = (String.IsNullOrEmpty(tabItem.Title) == false);
+            var hasIconURL = (String.IsNullOrEmpty(tabItem.IconURL) == false);
+
+            // Adds a new TabButton and Content View only if TabItem has TabButton properties and Content View.
+            if ((hasTitle || hasIconURL) && (tabItem.Content != null))
+            {
+                var tabButton = new TabButton(tabItem.TabButtonStyle);
```
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/TabView.cs:269]: Please suppress the warning complaining about disposing `TabButton`.

Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/TabView.cs:275]: Thank you for your comment!
I have added the SuppressMessage.

File: src/Tizen.NUI.Components/Controls/TabView.cs [hunk 2]
```diff
@@ -252,6 +252,38 @@ public void RemoveTab(int index)
...
+        /// <summary>
+        /// Adds a tab from the given TabItem.
+        /// TabItem contains Title and IconURL of a new TabButton in TabBar and Content of a new View in TabContent.
+        /// <param name="tabItem">The tab item which contains Title and IconURL of a new TabButton in TabBar and Content of a new View in TabContent.</param>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual void Add(TabItem tabItem)
+        {
+            var hasTitle = (String.IsNullOrEmpty(tabItem.Title) == false);
+            var hasIconURL = (String.IsNullOrEmpty(tabItem.IconURL) == false);
```
[the remark this answers is not in the document]
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/TabView.cs]: Thank you for  your comment!
I have changed IconURL to IconUrl.

File: src/Tizen.NUI.Components/Controls/TabItem.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public string Title { get; set; }
+
+        /// <summary>
+        /// IconURL of TabItem.
+        /// IconURL is set to the IconURL of a new TabButton in TabView's TabBar.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public string IconURL { get; set; }
```
[the remark this answers is not in the document]
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/TabItem.cs]: Thank you for  your comment!
I have changed IconURL to IconUrl.

**Review by rabbitfor — APPROVED** [review]
Thank you :)

**Pull request merged**